packages/docs-infra/src/pipeline/loaderUtils/fileUrlToPortablePath.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Converts a file:// URL to a portable path format that can be used with path-module (POSIX-only).
+ *
+ * This function is designed to work with isomorphic code that uses path-module,
+ * which only supports POSIX paths. The key insight is that by stripping the `file://`
+ * prefix and normalizing backslashes to forward slashes, we get a path that:
+ * - On Unix: `/home/user/file.ts` - works directly with path-module
+ * - On Windows: `/C:/Users/file.ts` - also works with path-module because it starts with `/`
```

Comment:
Don't we use `path-module` only as a browser shim? It feels wrong to me that it would ever receive absolute paths witha  drive letter, regardless of the platform. I'd expect them to be project-relative at least.